module_utils/foreman_helper.py

```diff
@@ -4,6 +4,23 @@
 import re
 import yaml
 
+from ansible.module_utils.basic import AnsibleModule
+
+
+class ForemanAnsibleModule(AnsibleModule):
+
+    def __init__(self, **kwargs):
+        custom_argument_spec = kwargs.pop('argument_spec', dict())
+        argument_spec = dict(
+            server_url=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+        )
+        argument_spec.update(custom_argument_spec)
+        kwargs['argument_spec'] = argument_spec
```

Comment:
Why don't you just update the `kwargs['argument_spec']` in place?

Reply:
because `kwargs['argument_spec'].update(argument_spec)` would override the values from the module with the defaults, and I couldn't implement `state` as discussed above (`ForemanAnsibleModule` has state `['present', 'absent']` and some modules add `present_with_defaults` and similar)

Reply:
Sometimes it's hard to see, where the changeable datatypes make a difference in python...